### DISCUSSION TITLE
Migration from legacy HA Core Integration to the new Custom Integration?

### DISCUSSION THREAD

**Discussion by mrbung0815** [SukramJ/aiohomematic#881] [Q&A] [answered]
Hello,

thx for the great work!

I have used the legacy HA Core Integration to implement Homematic CCU3 and around 100 HMIP devices.
Since i bought V2 devides of HMIP i figured out that i have to use the new custom integration.
Maybe it was kind of naiv but i hoped that the legacy entities which have been renamed and are configured in automations/dashboards would be matched with the entities imported by the new integration.
Unfortunately this did no happen :(

Is there a way to do the match manually?

Or do I need to rename all newly imported entities and reconfigure automations/dashboards?

BR
Stefan

**Comment by ThomDietrich** [comment, marked as answer]
Hey @mrbung0815,
Sorry to say but you will have to invest a bit of work. May I suggest (from experience) to follow a rather clean approach to this:

1. Delete the legacy integration and all orphaned entities left over in HA (restart HA)
2. Go into your Homematic UI and rename all devices here. Find a systematic naming so all devices follow the same pattern
3. Set up the new HA integration
4. Go through your automations and dashboards to see what needs to be updated to match the new devices. Not only because of names, but also to tender for changed behavior, e.g. of thermostats.

No surprises. The whole process will take you 1-2 hours and you will end up with a clean and consistent setup. Don't waste time and try to hold on to your "legacy" names with the sacrifice of a clean restart.

**Reply by mrbung0815** [reply]
Danke Dir für die Antwort.
Klingt nach nem Plan!
Gruß
Stefan

**Reply by ThomDietrich** [reply]
Läuft ;)

**Comment by emufan** [comment]
Also ich habe es anders gemacht und sollte in Deinem Fall vielleicht einfach sein.

Ich habe damals einfach die alte Integration gelöscht und eben gerade **nicht** die alten Entitäöten. Da die neue Integration zu 95% die gleichen IDs gleich verwendet und sie nach Installation dann "wieder da" sind und man so ebenfalls 95% der Automatisierungen, usw, die ja auf die entity_id aufsetzen dann weiterfunktionieren.
